platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkJavaCompiler.java

```diff
@@ -34,32 +39,93 @@
 import java.nio.charset.Charset;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
 import java.util.Set;
 
 public class JdkJavaCompiler implements Compiler<JavaCompileSpec>, Serializable {
     private static final Logger LOGGER = LoggerFactory.getLogger(JdkJavaCompiler.class);
 
     private final Factory<JavaCompiler> javaHomeBasedJavaCompilerFactory;
+    private final Problems problems;
 
     @Inject
-    public JdkJavaCompiler(Factory<JavaCompiler> javaHomeBasedJavaCompilerFactory) {
+    public JdkJavaCompiler(Factory<JavaCompiler> javaHomeBasedJavaCompilerFactory, Problems problems) {
         this.javaHomeBasedJavaCompilerFactory = javaHomeBasedJavaCompilerFactory;
+        this.problems = problems;
     }
 
     @Override
     public WorkResult execute(JavaCompileSpec spec) {
         LOGGER.info("Compiling with JDK Java compiler API.");
 
         ApiCompilerResult result = new ApiCompilerResult();
-        JavaCompiler.CompilationTask task = createCompileTask(spec, result);
+        DiagnosticCollector<JavaFileObject> diagnosticCollector = new DiagnosticCollector<>();
+        JavaCompiler.CompilationTask task = createCompileTask(spec, result, diagnosticCollector);
         boolean success = task.call();
+        reportProblems(diagnosticCollector);
         if (!success) {
             throw new CompilationFailedException(result);
         }
         return result;
     }
 
-    private JavaCompiler.CompilationTask createCompileTask(JavaCompileSpec spec, ApiCompilerResult result) {
+    private void reportProblems(DiagnosticCollector<JavaFileObject> diagnosticCollector) {
+        for (Diagnostic<? extends JavaFileObject> diagnostic : diagnosticCollector.getDiagnostics()) {
+            reportProblem(diagnostic);
+        }
+    }
+
+    private void reportProblem(Diagnostic<? extends JavaFileObject> diagnostic) {
+        String message = diagnostic.getMessage(Locale.getDefault());
+
+        String label = mapKindToLabel(diagnostic.getKind());
+        String resourceName = diagnostic.getSource().getName();
+        Integer line = Math.toIntExact(diagnostic.getLineNumber());
+        Integer column = Math.toIntExact(diagnostic.getColumnNumber());
+        Severity severity = mapKindToSeverity(diagnostic.getKind());
+
+        problems.createProblem(problem -> problem
+            .label(label)
+            .undocumented()
+            .location(resourceName, line, column)
+            .category("java", "compilation")
```

Comment:
💭 Is this category something general and it should be a constant?

Reply:
@reinsch82, did we eventually had a place, where we would standardize some categories?
How would you approach this?